src/app/backend/handler/apihandler.go

```diff
@@ -2156,14 +2160,39 @@ func (apiHandler *APIHandler) handleLogs(request *restful.Request, response *res
 		}
 	}
 
-	result, err := container.GetPodLogs(k8sClient, namespace, podID, containerID, logSelector)
+	result, err := container.GetLogDetails(k8sClient, namespace, podID, containerID, logSelector)
 	if err != nil {
 		handleInternalError(response, err)
 		return
 	}
 	response.WriteHeaderAndEntity(http.StatusOK, result)
 }
 
+func (apiHandler *APIHandler) handleLogFile(request *restful.Request, response *restful.Response) {
+	k8sClient, err := apiHandler.cManager.Client(request)
+	if err != nil {
+		handleInternalError(response, err)
+		return
+	}
+	namespace := request.PathParameter("namespace")
+	podID := request.PathParameter("pod")
+	containerID := request.PathParameter("container")
+
+	filename := fmt.Sprintf("attachment; filename='logs-from-%v-in-%v.txt'", containerID, podID)
```

Comment:
Yes, I was not happy with the implementation, too. Currently, apihandler is (mostly) dispatching. This method adds some logic and bloats apihandler even more.

On the other hand, adding the http response object as an input parameter of the business logic is not nice, either. 

Maybe there is another option: what about creating a generic download handler and put it to the handler package?